msprime/provenance.py

```diff
@@ -113,9 +113,7 @@ def default(self, obj):
             }
 
         if isinstance(obj, tskit.TreeSequence):
-            # Store the tree sequence as the tables deserialised with
-            # tskit.TableCollection.fromdict(tables).tree_sequence()
-            return {'__ts__': obj.tables.asdict()}
+            return "__current_ts__"
```

Comment:
I think `{'__constant__': '__current_ts__'}` would be better here like the other "interpretable" parts, this way it doesn't get mixed up with strings.